client/my-sites/upgrades/domain-management/components/domain/maintenance-card.jsx

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import React from 'react' ;
+
+/**
+ * Internal dependencies
+ */
+import { localize } from 'i18n-calypso';
+import { getSelectedDomain } from 'lib/domains';
+import Notice from 'components/notice';
+
+const MaintenanceCard = ( { domains, selectedDomainName, translate } ) => {
+	const domain = getSelectedDomain( { domains, selectedDomainName } ),
+		lastIndexOfDot = domain.name.lastIndexOf( '.' ),
+		tld = lastIndexOfDot !== -1 && domain.name.substring( lastIndexOfDot );
```

Comment:
We have this code duplicated now in three places:
* [components/domains/map-domain-step/index.jsx](https://github.com/Automattic/wp-calypso/blob/edec8ce7aa72e1218cfd86f1d2b861aa72b3edb3/client/components/domains/map-domain-step/index.jsx#L189-L190)
* [components/domains/register-domain-step/index.jsx](https://github.com/Automattic/wp-calypso/blob/9e47edc5ab546054e125fee321f63f7de239e1a0/client/components/domains/register-domain-step/index.jsx#L552-L553)
* here

I think it's high time we extract it to a helper function, probably in `lib/domains` or something :)